cmd/collector/app/flags/flags.go

```diff
@@ -106,71 +115,33 @@ type CollectorOptions struct {
 	// NumWorkers is the number of internal workers in a collector
 	NumWorkers int
 	// HTTP section defines options for HTTP server
-	HTTP HTTPOptions
+	HTTP confighttp.ServerConfig
 	// GRPC section defines options for gRPC server
-	GRPC GRPCOptions
+	GRPC configgrpc.ServerConfig
 	// OTLP section defines options for servers accepting OpenTelemetry OTLP format
 	OTLP struct {
 		Enabled bool
-		GRPC    GRPCOptions
-		HTTP    HTTPOptions
+		GRPC    configgrpc.ServerConfig
+		HTTP    confighttp.ServerConfig
 	}
 	// Zipkin section defines options for Zipkin HTTP server
 	Zipkin struct {
-		// HTTPHostPort is the host:port address that the Zipkin collector service listens in on for http requests
-		HTTPHostPort string
-		// TLS configures secure transport for Zipkin endpoint to collect spans
-		TLS tlscfg.Options
-		// CORS allows CORS requests , sets the values for Allowed Headers and Allowed Origins.
-		CORS corscfg.Options
-		// KeepAlive configures allow Keep-Alive for Zipkin HTTP server
-		KeepAlive bool
+		confighttp.ServerConfig
+		KeepAlive bool `mapstructure:"keep-alive"` // Add keepalive setting
```

Comment:
why are we defining a mapstructure tag for this? I actually did a search in the codebase and it doesn't look like this is being used anywhere - can we just remove it? cc @yurishkuro

Reply:
+1